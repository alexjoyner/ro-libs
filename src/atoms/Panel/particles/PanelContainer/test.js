import React from 'react';
import { shallow } from 'enzyme';
import { PanelContainer } from './';

describe('PanelContainer component', () => {
  let component;
  beforeEach(() => {
    component = shallow(<PanelContainer />);
  });
  it('Should render without exploding', () => {
    expect(component).toBeDefined();
  });
  it('Should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
  it('should have correct default styles', () => {
    expect(component).toHaveStyleRule('display', 'block');
    expect(component).toHaveStyleRule('min-width', '280px');
  });
  it('should apply correct min width', () => {
    const minWidthTest = shallow(<PanelContainer minWidth="300px" />);
    expect(minWidthTest).toHaveStyleRule('min-width', '300px');
  });
  it('should apply correct width styles', () => {
    const widthTest = shallow(<PanelContainer width="600px" />);
    expect(widthTest).toHaveStyleRule('width', '600px');
    expect(widthTest).toHaveStyleRule('display', 'inline-block');
  });
});
