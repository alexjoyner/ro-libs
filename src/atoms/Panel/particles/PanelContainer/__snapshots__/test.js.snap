// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelContainer component Should match snapshot 1`] = `
.c0 {
  display: block;
  position: relative;
  height: auto;
  min-width: 280px;
  margin: 10px;
  padding: 10px;
  background-color: #fafafa;
  border: 1px solid #EEEEEE;
  radius: 5px;
  box-shadow: 0 2px 2px 0 rgba(0,0,0,0.16);
}

<styled.div>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
    ",
            [Function],
            "
    position: relative;
    height: ",
            [Function],
            ";
    min-width: ",
            [Function],
            ";
    margin: 10px;
    padding: 10px;
    background-color: ",
            "#fafafa",
            ";
    border: 1px solid #EEEEEE;
    radius: 5px;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.16);
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</styled.div>
`;
